peg_merge/views.py

```diff
@@ -0,0 +1,65 @@
+import hmac
+
+from django.conf import settings
+from django.contrib.auth import logout
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.contrib.auth.models import User
+from django.db import transaction
+from django.urls import reverse
+from django.utils.encoding import force_bytes
+from django.utils.translation import gettext as _
+from django.views.generic import FormView
+from peg_merge.forms import WCIPEGMergeActivationForm, WCIPEGMergeRequestForm
+
+from judge.models import Comment, CommentVote, Profile, Submission
+from judge.utils.views import TitleMixin, generic_message
+
+
+class WCIPEGMergeRequest(LoginRequiredMixin, UserPassesTestMixin, TitleMixin, FormView):
+    template_name = 'user/wcipeg-merge-request.html'
+    form_class = WCIPEGMergeRequestForm
+
+    def test_func(self):
+        return self.request.profile.is_peg
+
+    def get_title(self):
+        return _('WCIPEG Merge Request')
+
+    def form_valid(self, form):
+        user = User.objects.get(username=form.cleaned_data['handle'])
+        token = 'wcipeg_%d_%s' % (user.pk, self.request.user.email)
```

Comment:
Is this email correct? In both views, we're using `self.request.user.email`, but the user changes between merge request and merge activate.

Reply:
It doesn't. The WCIPEG user only gets logged out once the migration is finished.

Reply:
Wait, so you are confirming the merge from the PEG account?

Reply:
Yes. The whole idea is that we verify PEG with password and verify DMOJ with email.